File: PetFinderApp/app/src/main/java/com/example/petfinderapp/model/Publicacao.java
package com.example.petfinderapp.model;

public class Publicacao {

    private long id;
    private String descricao;
    private String nomePet;
    private String genero;
    private String especie;
    private String porte;
    private String idade;
    private String vacinas;
    private boolean Castrado;
    private String imagem;
    private long userId;

    public Publicacao(String descricao, String nomePet, String genero, String especie, String porte, String idade, String vacinas, boolean castrado, long userId) {
        this.descricao = descricao;
        this.nomePet = nomePet;
        this.genero = genero;
        this.especie = especie;
        this.porte = porte;
        this.idade = idade;
        this.vacinas = vacinas;
        this.Castrado = castrado;
        this.userId = userId;
    }

    public Publicacao(String descricao, String nomePet, String genero, String especie, String porte, String idade, String vacinas, boolean castrado, String imagem, long userId) {
        this.descricao = descricao;
        this.nomePet = nomePet;
        this.genero = genero;
        this.especie = especie;
        this.porte = porte;
        this.idade = idade;
        this.vacinas = vacinas;
        this.Castrado = castrado;
        this.imagem = imagem;
        this.userId = userId;
    }

    public Publicacao(long id, String descricao, String nomePet, String genero, String especie, String porte, String idade, String vacinas, boolean castrado, String imagem, long userId) {
        this.id = id;
        this.descricao = descricao;
        this.nomePet = nomePet;
        this.genero = genero;
        this.especie = especie;
        this.porte = porte;
        this.idade = idade;
        this.vacinas = vacinas;
        this.Castrado = castrado;
        this.imagem = imagem;
        this.userId = userId;
    }

    public long getId() {
        return id;
    }

    public String getDescricao() {
        return descricao;
    }

    public String getNomePet() {
        return nomePet;
    }

    public String getGenero() {
        return genero;
    }

    public String getEspecie() {
        return especie;
    }

    public String getPorte() {
        return porte;
    }

    public String getIdade() {
        return idade;
    }

    public String getVacinas() {
        return vacinas;
    }

    public boolean isCastrado() {
        return Castrado;
    }
    public String getImagem() {
        return imagem;
    }

    public long getUserId() {
        return userId;
    }
}

File: backend/petFinder-app/app/Models/Publicacao.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Concerns\HasTimestamps;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Publicacao extends Model
{
    use HasFactory, HasTimestamps;

    protected $table = 'publicacoes';
    protected $hidden = ['image_path'];
    
    public function user()
    {
        return $this->belongsTo(User::class, 'user_id')->select('id', 'name', 'avatar', 'genero', 'dataNasc', 'telefone');
    }
   
    public function favoritos()
    {
        return $this->belongsToMany(User::class, 'favoritos', 'publicacao_id', 'user_id');
    }

    protected $appends = ['image_link'];
    public function getImageLinkAttribute()
    {
        return asset($this->image_path);
    }

    protected $fillable = [
        'nomePet',
        'porte',
        'idade',
        'vacinas',
        'castrado',
        'genero',
        'especie',
        'descricao',        
        'image_path',
        'user_id',
    ];
}
